parquet-avro/src/main/java/org/apache/parquet/avro/AvroRecordConverter.java

```diff
@@ -194,12 +194,12 @@ private static void addLogicalTypeConversion(SpecificData model, Schema schema,
                 model.addLogicalTypeConversion(conversion);
               }
             }
-
+          } catch (NoSuchFieldException e) {
+            // Avro classes without logical types (denoted by the "conversions" field)
+          } finally {
```

Comment:
I don't think this should be in the finally block. We should not execute that in case of other exception

Reply:
hm what should we do instead? Just because the `conversions` class is null (meaning no top-level logical types) doesn't mean we shouldn't traverse the inner fields for any record types with conversions... at least in the generated classes I looked at

Reply:
just have it after the try/catch without enclosing finally

Reply:
👍 done